### PR TITLE
fix(azure-deployment): Trigger repull on new catalogue version by upd…

### LINKED ISSUE #136: 🐞 BUG REPORT
### 🔄 Duplicate Check

- [X] I have searched both open and closed issues and found no similar reports.

### 🔍 What Happened?

When there is a new version of the catalogue, it does not trigger a repull on the Azure site because the image name remains unchanged

### 👣 Steps to Reproduce

1. Trigger main build with visible changes
2. Go to https://catalog-test.healthdata.nl/
3. The changes are visible

### 📸 Screenshots

_No response_

### 📋 Logs

_No response_

### 🌐 Affected Browsers

_No response_

### 💻 Operating System

_No response_

### 🤝 Want to Contribute to the Fix?

Yes, I'd like to help fix this.

### ℹ️ Additional Information

_No response_

### PR DESCRIPTION
**Pull request by hcvdwerf**
…ating image closes #136

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -88,9 +56,19 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           push: true
-          tags: ${{ steps.meta.outputs.tags }}
+          tags: "${{ env.IMAGE_NAME }}:${{ github.sha }}"
           labels: ${{ steps.meta.outputs.labels }}
 
+      - name: Run Trivy vulnerability scanner
```
Comment (brunopacheco1) [inline .github/workflows/main.yml:62]: important: moving Trivy scanner to this part of the pipeline, will not prevent a vulnerable docker image to be uploaded into the registry. I strongly recommend you execute trivy on a locally build image before pushing it.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -88,9 +56,19 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           push: true
-          tags: ${{ steps.meta.outputs.tags }}
+          tags: "${{ env.IMAGE_NAME }}:${{ github.sha }}"
```
Comment (brunopacheco1) [inline .github/workflows/main.yml:59]: important: I strongly recommend you to revert this change.

`github.sha` is a commit id, which mean you are going to push to the repository always new tags, but `steps.meta.outputs.tags` is a variable filled by `docker/build-push-action` that conveys to tags according to the branch/release strategy we follow, please check their docs.

Besides that, a tag should follow https://semver.org/, but `SHA` is not readable and gives us no context at all.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -101,4 +79,4 @@ jobs:
         with:
           app-name: ${{ env.AZURE_WEBAPP_NAME }}
           publish-profile: ${{ secrets.AZURE_WEBAPP_PUBLISH_PROFILE }}
-          images: "${{ env.IMAGE_NAME }}:main"
+          images: "${{ env.IMAGE_NAME }}:${{ github.sha }}"
```
Comment (brunopacheco1) [inline .github/workflows/main.yml:82]: can you please check an alternative way to redeploy the latest version of a tag? Why is it not happening for the frontend?

**Review by brunopacheco1 — CHANGES_REQUESTED** [review]
can you please check an alternative way to redeploy the latest version of a tag? Why is it not happening for the frontend?

File: .github/workflows/main.yml (hunk 2 above)
Comment (brunopacheco1) [reply .github/workflows/main.yml:61]: I remember having this issue with Amazon and Kubernetes, but we solved it with labels. The tag was still the same, but every new push, we set a label with the commit it. Perhaps Azure also supports it.

In the worse case we can push two tags, the correct one and the commit id tag, but I would recommend you to try first other alternatives. Would you mind checking it?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -29,76 +29,76 @@ jobs:
...
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=sha
 
-      - name: Build Docker image for scanning
+      - name: Extract second tag
```
[the remark this answers is not in the document]
Comment (hcvdwerf) [reply .github/workflows/main.yml]: Extract last tag from the list always

**Review by brunopacheco1 — APPROVED** [review]

**Pull request merged**